instrumentation/http_client/lib/opentelemetry/instrumentation/http_client/patches/session.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module HttpClient
+      module Patches
+        # Module to prepend to HTTPClient::Session for instrumentation
+        module Session
+          def connect
+            site = @proxy || @dest
+            url = site.addr
+
+            attributes = OpenTelemetry::Common::HTTP::ClientContext.attributes.merge('http.url' => url)
+            tracer.in_span('HTTP CONNECT', attributes: attributes, kind: :client) do
+              super
+            end
```

Comment:
There's some weirdness with this instrumentation:
1. We're not injecting context. On one hand, this is probably OK because the server side probably isn't tracing connection establishment, on the other hand, why are we calling this a `client` span?
2. *If* the server *is* instrumenting connection establishment, then the `server` span(s) will be associated with the wrong `client` span, because a request span is the parent of this span, and it *does* inject context.

Reply:
This is a shameless port of our internal instrumentation of this gem, weirdness and all 😄 .

Reply:
meh, ok 😉

Reply:
Removed client span type from connect span.